tests/python/gpu/test_tvm_bridge.py

```diff
@@ -0,0 +1,63 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Test TVM bridge, only enable this when TVM is available"""
+import mxnet as mx
+import numpy as np
+
+def test_tvm_bridge():
+    # only enable test if TVM is available
+    try:
+        import tvm
+        import tvm.contrib.mxnet
+        import topi
+    except ImportError:
+        print("TVM bridge test skipped because TVM is missing...")
+        return
+
+    def check(target):
+        shape = (20,)
+        scale = tvm.var("scale", dtype="float32")
+        x = tvm.placeholder(shape)
+        y = tvm.placeholder(shape)
+        z = tvm.compute(shape, lambda i: x[i] + y[i])
+        zz = tvm.compute(shape, lambda *i: z(*i) * scale)
+        ctx = mx.gpu(0) if target == "cuda" else mx.cpu(0)
+        target = tvm.target.create(target)
+
+        # build the function
+        with target:
+            s = topi.generic.schedule_injective(zz)
+            f = tvm.build(s, [x, y, zz, scale])
+
+        # get a mxnet version
+        mxf = tvm.contrib.mxnet.to_mxnet_func(f, const_loc=[0, 1])
```

Comment:
This design feels weird. MXNet and TVM are mutually referencing each other.
mxnet backend is referencing TVM with TVMBridge and tvm is referencing mxnet with contrib.mxnet

Reply:
The benefit of this design, is that MXNet do not have to link libtvm.so which comes with llvm dependency and it will work out of box when mxnet and tvm are both available

Reply:
Logically, we can view this as both tvm and mxnet depend on header only fraction of TVM runtime, and TVM contrib library calls MXNet to retrieve the bridge function

Reply:
Sorry to chime in here, was just wondering: Does TVM look for this function specifically, or does mxnet call a TVM function beforehand, passing it a pointer to this callback (which would seem less circular).

Reply:
TVM looks for this function specifically.